modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCSSOConsentTestCase.java

```diff
@@ -0,0 +1,343 @@
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.json.simple.JSONValue;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.common.model.xsd.Claim;
+import org.wso2.carbon.identity.application.common.model.xsd.ClaimConfig;
+import org.wso2.carbon.identity.application.common.model.xsd.ClaimMapping;
+import org.wso2.carbon.identity.application.common.model.xsd.ServiceProvider;
+import org.apache.commons.lang.StringUtils;
+import org.wso2.identity.integration.test.oidc.bean.OIDCApplication;
+import org.wso2.identity.integration.test.oidc.bean.OIDCUser;
+import org.wso2.identity.integration.test.util.Utils;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Test case to test the consent management with OIDC SSO.
+ */
+public class OIDCSSOConsentTestCase extends OIDCAbstractIntegrationTest {
+
+    protected OIDCUser user;
+    protected String accessToken;
+    protected String sessionDataKeyConsent;
+    protected String sessionDataKey;
+    protected String authorizationCode;
+
+    CookieStore cookieStore = new BasicCookieStore();
+
+    protected HttpClient client;
+    protected List<NameValuePair> consentParameters = new ArrayList<>();
+    OIDCApplication playgroundApp;
+    ServiceProvider serviceProvider;
+    private String claimsToGetConsent;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+
+        initUser();
+        createUser(user);
+
+        playgroundApp = initApplication();
+        serviceProvider = createApplication(new ServiceProvider(), playgroundApp);
+        client = HttpClientBuilder.create().setDefaultCookieStore(cookieStore).build();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testClear() throws Exception {
+
+        deleteUser(user);
+        deleteApplication(playgroundApp);
+        clear();
+    }
+
+    @Test(groups = "wso2.is", description = "Test consent management after updating " +
+            "the application claim configurations")
+    public void testConsentWithAppClaimConfigUpdate() throws Exception {
+
+        testSendAuthenticationRequest(playgroundApp, client);
+        testAuthentication(playgroundApp);
+        Assert.assertEquals(claimsToGetConsent, "0_Email,1_First Name",
+                "Incorrect claims to request the consent");
```

Comment:
Can improve the messaging to be more clear on test fail. Sentense case followed in the failure message needs the full stop at the end. Let's correct other places as well. :)

Reply:
Fixed https://github.com/wso2/product-is/pull/11407/commits/4fe18ee0052418875fffa6a22be116143ea73ebb